sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -43,6 +43,9 @@ public abstract class EventProcessor<TPartition> where TPartition : EventProcess
         /// <summary>The maximum number of failed consumers to allow when processing a partition; failed consumers are those which have been unable to receive and process events.</summary>
         private const int MaximumFailedConsumerCount = 1;
 
+        /// <summary>Indicates whether or not the consumer should consider itself invalid when a partition is stolen by another consumer, as determined by the Event Hubs service.</summary>
+        private const bool InvalidateConsumerWhenPartitionIsStolen = true;
```

Comment:
I see this defaulted to false elsewhere. Is that expected?

Reply:
Yup;  we want the other consumer and receiver clients to fail if they ask to read events so they see the background exception.   The caller can make the decision to invoke again which will reassert control when competing consumers have the same owner level.   _(in the future, we may expose this as an option for the other types)_

In this case, we want the partition processor to fault and not have a chance to recover with retries so that it immediately respects the ownership change for the partition.  In the worst case, when load balancing runs again, if storage says this processor still owns the partition, a new partition processing task will get spun up and reassert ownership.

In truth, it's a bit of overkill, as we don't normally retry this type of exception, but we have the mechanism in place and using it makes sense.